### PR TITLE
Pulls changes from upstream for v1.4.0.0-PN-SNAPSHOT

### PR DESCRIPTION
**Pull request by joserobjr**
This is a continuation of #970 but targeting 1.4.0.0-PN-SNAPSHOT, this will also sync with #990, keeping our project in sync with Cloudburst Nukkit.

----
## Current Status:
Needs to test all the blocks that have a GUI

---
## Needs to check
GUI:
- [x] Creative inventory
- [x] Survival inventory
- [x] Survival crafting grid (2x2)
- [x] Adventure inventory
- [x] Adventure crafting grid (2x2)
- [x] Crafting table
- [x] Cartography table
- [x] Smithing table
- [x] Furnace
- [x] Blast furnace
- [x] Smoker
- [x] Brewing stand
- [x] Anvil
- [x] Grindstone
- [x] Enchanting table
- [x] Lectern
- [x] Chest
- [x] Barrel
- [x] Shulker Box
- [x] Sign
- [x] Item Frame
- [x] Beacon
- [x] Stonecutter
- [x] Hopper
- [x] Dropper
- [x] Dispenser
- [x] Loom

Blocks:
- [x] If breaking ice block (all types) with silk touch matches the vanilla behaviour
- [x] The lava flow in overworld and in the nether
- [x] If minecraft:stripped_crimson_hyphae, minecraft:stripped_crimson_stem, minecraft:stripped_warped_hyphae, minecraft:stripped_warped_stem, are OK

Entities:
- [x] If we can still name the entities with name tag (and only one item is consumed from a stack in survival)
- [x] If the nametag behaviour with armor stands matches vanilla (and the same item consumption test)
---
## Issues found:

- [x] Can't break dragon eggs in creative mode https://discord.com/channels/728280425255927879/728280425255927882/820927345337827390
- [x] Skeleton and Wither Skeleton spawn eggs aren't working correctly
- [x] Grindstone opens GUI but it won't work
- [x] `/give @p nametag` shows as possibility but don't work, only `name_tag` works
- [x] Netherite items vibrates and desyncs when thrown in lava (should be handled as a bug later)
- [x] Secret issue1
- [x] Brewing stand inventory slots works but the brewing process won't start (like if the recipe was not valid)
- [x] stonecuttern is not showing any possible cut
- [x] Dispensing a shulkerbox in a disperser deletes the shulkerbox

---
## Not implemented:
- Cartography table
- Smithing table #648
- Loom
- Furnace, Blast Furnace, and Smoker are not giving exp when taking the item

### CODE REVIEW

**Review by joserobjr — COMMENTED** [review]
Self code review.
Also need to revise the comments on #970

**Review by joserobjr — COMMENTED** [review]
Self reviewed and no problems were found. Moving on...

**Review by joserobjr — COMMENTED** [review]
No more issues found on self-review, moving on...

File: src/main/java/cn/nukkit/network/protocol/AvailableCommandsPacket.java [hunk 1]
```diff
@@ -31,17 +31,14 @@
...
-    public static final int ARG_TYPE_JSON = 47;
-    public static final int ARG_TYPE_COMMAND = 54;
+    public static final int ARG_TYPE_FILE_PATH = 15;
+    public static final int ARG_TYPE_STRING = 31;
+    public static final int ARG_TYPE_BLOCK_POSITION = 39;
+    public static final int ARG_TYPE_POSITION = 40;
+    public static final int ARG_TYPE_MESSAGE = 43;
+    public static final int ARG_TYPE_RAWTEXT = 45;
+    public static final int ARG_TYPE_JSON = 49;
+    public static final int ARG_TYPE_COMMAND = 56;
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/network/protocol/AvailableCommandsPacket.java:41]: Breaking change

**Review by joserobjr — COMMENTED** [review]
Self-reviewed and found an issue but it is already being handled at #990

**Review by joserobjr — COMMENTED** [review]
Self-reviewed and no issues were found, continuing...

**Review by joserobjr — COMMENTED** [review]
Self-reviewed and no issues found. Running tests in-game.

**Review by joserobjr — COMMENTED** [review]
Self-reviewed and tested in game. Merging...

**Comment by Jl1mbo** [comment]
brewing stand don't work

**Comment by Jl1mbo** [comment]
lol

**Comment by RealTriassic** [comment]
lol

**Comment by MonoAdrian23** [comment]
Looms and Cartography Tables doenst work

**Comment by TheDiego845** [comment]
Grindstone doesn't work either

**Pull request merged**